packages/gatsby-source-filesystem/src/create-file-node.js

```diff
@@ -7,21 +7,14 @@ const prettyBytes = require(`pretty-bytes`)
 const md5File = require(`bluebird`).promisify(require(`md5-file`))
 const crypto = require(`crypto`)
 
-exports.createFileNode = async (
-  pathToFile,
-  createNodeId,
-  pluginOptions = {}
-) => {
+exports.createFileNode = async (pathToFile, createNodeId, pluginName) => {
   const slashed = slash(pathToFile)
   const parsedSlashed = path.parse(slashed)
   const slashedFile = {
     ...parsedSlashed,
     absolutePath: slashed,
     // Useful for limiting graphql query with certain parent directory
-    relativeDirectory: path.relative(
-      pluginOptions.path || process.cwd(),
-      parsedSlashed.dir
-    ),
+    relativeDirectory: path.relative(process.cwd(), parsedSlashed.dir),
```

Comment:
i'm unsure if this breaks anything but the conditional here was always true so this is now a path that wasn't hit before

Reply:
yeah, this is definitely breaking change (at very least e2e tests for gatsby-image rely on current behaviour - well for the `relativePath` which was changed similarly).

currently `process.cwd` path is hit/used by `createRemoteFileNode` utility, where there are no pluginOptions (just for context on why it's here)

Reply:
Ah ok, I'm not really sure I understand what the old behavior was doing/what it was for so I'm not sure if we can work around it. The simplest back compatible change tho would be to keep the old behavior if options.path is a single non glab path. 

Again tho I don't really understand what this is doing so maybe that still breaks a bunch of usage for folks that opt into this newer API path?

Reply:
for the most part it was just setting `relative(Directory|Path)` fields as path relative to one passed in plugin options:

if we had:
```
{
  resolve: `gatsby-source-filesystem`,
  options: {
    name: `content`,
    path: `${__dirname}/src/content/`,
  }
}
```

then `src/content/blog/blogpost_1/index.md` would have `relativePath` as `blog/blogpost_1/index.md` (instead of path relative to project root).

we definitely can't work it out if `path` passed to pluginOptions is not straight path to directory, but is some kind of glob pattern, but I do feel like current behaviour is weird - especially if you have multiple instances of filesystem plugin, each pointing to different directory - then relativePath in global context is really hard to reason with

Reply:
what is the value in the relative path here? It seems like it avoids situations where it's sourced from outside the `cwd`? I'd agree on the face of it it's weird that "relativePath" not being anchored always to the site root seems confusing. Maybe there is a better way to handle whatever usecase this is enabling?